### ISSUE TITLE
E-Voting (Pairwise comparison) flow

### ISSUE THREAD

**Issue by MoeNick** [GeneralMagicio/pairwise-RPGF4#21]
As a user I want to compare my selected projects in pairs so I can have a better curation and rankings.
<img width="669" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBLzdSQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--42bc6088d8a2345dd141994316ce10c470fb5ba4/image.png" alt="image.png" />

**Success Criteria**

- [x] Progress bar is % of total possible pair -voting which is n * (n- 1) / 2 which is n is the total accepted/filtered projects in a category
- [x] by a threshold passed, a modal asks user to continue voting or see the rankings
 <img width="273" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBd1BTQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1cf73a6903453a4915be8435931e250d27338c9e/image.png" alt="image.png" /> 

[link to design](https://www.figma.com/design/VeVWDkxYhD8G3AxMFiFYp3/Pairwise-Mobile-Voting-App?node-id=2154-4176&t=nPvyaLabrEbsywcB-4)

- [x] the threshold is 0.2 of the above formula. 
- [x] if the number of selected projects is less than 7, then the threshold should be 0.4
- [ ] #66

- [x] the finish ranking button will appear at the  button of screen if user passed the threshold and stll want to do ranking.
- [x] clicking on finish ranking will show the "ranking is done page".

**Comment by MoeNick** [comment]
@mmahdigh @alireza-sharifpour the success criteria is updated.

**Comment by alireza-sharifpour** [comment]
It's ready for testing @maryjaf @MoeNick

**Comment by maryjaf** [comment]
- [ ] 1- 

-I've selected 2 projects in a category and after completing ranking flow I see "start ranking" button again @alireza-sharifpour 

https://github.com/GeneralMagicio/pairwise-RPGF4/assets/111529185/e67acde1-061a-48ef-9a1b-ca91175bcc61

**Comment by maryjaf** [comment]
- [ ] 2-
When The user selects only one project in a category, after completing select project flow , loading icon is shown 
(I think this comment https://github.com/GeneralMagicio/pairwise-RPGF4/issues/19#issuecomment-2107587971 also related to this  )

**Comment by MoeNick** [comment]
@maryjaf please also take a look at the formula in the success criteria. maybe it's a condition that might be correct according the minimum threshold.

**Comment by maryjaf** [comment]
> @maryjaf please also take a look at the formula in the success criteria. maybe it's a condition that might be correct according the minimum threshold.

I think In this situation appropriate message should be shown not loading icon

**Comment by maryjaf** [comment]
- [ ] 3-
I’m not sure this problem is exactly relate to this issue or not
The images isn’t shown in ranking flow between  selected projects or categories
![IMG_1620](https://github.com/GeneralMagicio/pairwise-RPGF4/assets/111529185/6b95da90-0fc5-457f-9ab2-dad1bbb79263)
![IMG_1617](https://github.com/GeneralMagicio/pairwise-RPGF4/assets/111529185/c7c6377b-a812-4567-915d-aac3100d105c)

**Comment by maryjaf** [comment]
- [ ] 4-
By tapping on “continue ranking” the submit vote page is shown just like when user taps on finish ranking 
Please take a look on below video 

https://github.com/GeneralMagicio/pairwise-RPGF4/assets/111529185/2967af80-7453-434e-9ccf-05139b84b8d8

**Comment by alireza-sharifpour** [comment]
> * [ ]  1-
> 
> -I've selected 2 projects in a category and after completing ranking flow I see "start ranking" button again @alireza-sharifpour
> 
>  Screen.Recording.2024-05-14.at.11.04.43.AM.mov

This section only has implemented until the comment part. we don't have backend for after this flow.

**Comment by alireza-sharifpour** [comment]
> * [ ]  3-
>   I’m not sure this problem is exactly relate to this issue or not
>   The images isn’t shown in ranking flow between  selected projects or categories
>   ![IMG_1620](https://private-user-images.githubusercontent.com/111529185/330355168-6b95da90-0fc5-457f-9ab2-dad1bbb79263.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3MDAwODIsIm5iZiI6MTcxNTY5OTc4MiwicGF0aCI6Ii8xMTE1MjkxODUvMzMwMzU1MTY4LTZiOTVkYTkwLTBmYzUtNDU3Zi05YWIyLWRhZDFiYmI3OTI2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNFQxNTE2MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNWYwNmZhYTI5YjUwODhhYTZjYzhhZjUyZjcyNzgwMGI5ODRiOGI2MDc5MzU0MjBjOTBhYzRhNWY0NmE2NzYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1R9eR4a1E9izIcROty-btFd_Xt5E0BGgiGROVGPUGeY)
>   ![IMG_1617](https://private-user-images.githubusercontent.com/111529185/330355180-c7c6377b-a812-4567-915d-aac3100d105c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU3MDAwODIsIm5iZiI6MTcxNTY5OTc4MiwicGF0aCI6Ii8xMTE1MjkxODUvMzMwMzU1MTgwLWM3YzYzNzdiLWE4MTItNDU2Ny05MTVkLWFhYzMxMDBkMTA1Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxNFQxNTE2MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NTQ0N2Y4ODhmZjc0OWY4MTNhNTBmM2FhNzQzODAwYTRmZmJiMmRjNjA4ZGE5ZmNjM2Y1YWMyODExY2ZhNzBlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.U3sNiEbEKjUzdWsvmsL3X1q_Uh1yJao-LwgWanm99PU)

It can be related to your internet connection.

**Comment by maryjaf** [comment]
> > * [ ]   1-
> > 
> > -I've selected 2 projects in a category and after completing ranking flow I see "start ranking" button again @alireza-sharifpour
> > Screen.Recording.2024-05-14.at.11.04.43.AM.mov
> 
> This section only has implemented until the comment part. we don't have backend for after this flow.

So it should be fixed by backend ? @alireza-sharifpour @mmahdigh

**Comment by mmahdigh** [comment]
> > > * [ ]    1-
> > > 
> > > -I've selected 2 projects in a category and after completing ranking flow I see "start ranking" button again @alireza-sharifpour
> > > Screen.Recording.2024-05-14.at.11.04.43.AM.mov
> > 
> > 
> > This section only has implemented until the comment part. we don't have backend for after this flow.
> 
> So it should be fixed by backend ? @alireza-sharifpour @mmahdigh

After #44 is merged, Those categories will have the `Attested` status and then we can completely lock a category to allow no further editing.

**Comment by mmahdigh** [comment]
> * [ ]  3-
>   By tapping on “continue ranking” the ”Well Done” page and submit vote button are  shown just like when user taps on “finish ranking “
>   Please take a look on below video
>   It happens for me only on mobile @alireza-sharifpour
> 
>  RPReplay_Final1715679169.mp4

This is fixed. Please retest it.

**Comment by maryjaf** [comment]
> * By tapping on “continue ranking” the ”Well Done” page and submit vote button are  shown just like when user taps on “finish ranking “
>   Please take a look on below video

Thanks @mmahdigh item 3 has been resolved

**Comment by MoeNick** [comment]
Shall I move it to done @mmahdigh @maryjaf what's left?

**Comment by maryjaf** [comment]
> * selected projects has a an animation, you can see it here in [prototype](https://www.figma.com/proto/VeVWDkxYhD8G3AxMFiFYp3/Pairwise-Mobile-Voting-App?page-id=519%3A7426&type=design&node-id=1042-6702&viewport=932%2C-704%2C0.13&t=xxoBy1nwNS4NlUbD-1&scaling=scale-down&starting-point-node-id=718%3A3754)

This AC remains @MoeNick

**Comment by MoeNick** [comment]
Raised another issue and moved it to done
